src/protocol/data/resp/parse.c

```diff
@@ -81,67 +83,189 @@ _parse_cmd(struct request *req)
     return PARSE_OK;
 }
 
+static parse_rstatus_e
+_parse_range(struct array *token, struct buf *buf, int64_t nelem)
+{
+    parse_rstatus_e status;
+    struct element *el;
+
+    while (nelem > 0) {
+        if (buf_rsize(buf) == 0) {
+            return PARSE_EUNFIN;
+        }
+        el = array_push(token);
+        status = parse_element(el, buf);
+        log_verb("parse element returned status %d", status);
+        if (status != PARSE_OK) {
+            return status;
+        }
+        nelem--;
+    }
+
+    return PARSE_OK;
+}
+
+
+static void
+_parse_req_attrib(struct request *req, struct element *key, struct element *val)
+{
+    attrib_type_e type = ATTRIB_UNKNOWN;
+
+    /* treat this as it cannot fail, skip anything that is not recognized or has
+     * invalid value so the program will proceed to the request main itself.
+     */
+    if (key->type != ELEM_STR) { /* key has to be a simple string */
+        log_warn("attribute key must be simple string, not type %d", key->type);
+
+        return;
+    }
+    while (++type < ATTRIB_SENTINEL &&
```

Comment:
nit: imo this (and the following if clause) would be more readable as a for

Reply:
changed